src/exchange-connector/exchangeDownloader.ts

```diff
@@ -159,29 +160,33 @@ export function searchExchange(
  * @param {RegExp} deployment
  * @returns {Promise<string>} Returned the version string that matches the regex passed.  Will return first found result
  */
-export function getVersionByDeployment(
+export async function getVersionByDeployment(
```

Comment:
[PRAISE] thank you for refactoring! much easier to read

Reply:
🙏